### DISCUSSION TITLE
history not showing gitlab ci/cd

### DISCUSSION THREAD

**Discussion by fauzisunarya** [allure-framework/allure2#2427] [Questions & Support]
Hallo sir, please help me , i try implementation allure report in gitlab ci/cd, im having problems, trend history, duration history etc its not working,

**Comment by baev** [comment]
Could you please specify a bit more details on what was your steps & what you already tried?

**Comment by fauzisunarya** [comment]
hello sir, I am implementing an allure report for QA results using cypress, when I use github action and jenkins it works very well and displays testing history and trendlines graphs also appear automatically.

but when I implemented it using gitlab CI/CD history test it didn't work and there was no trendlines graph, do you have a solution so that allure report works perfectly in gitlab ci/cd, I read a lot of references that for implementation on gitlab ci/cd history it won't work requires a special configuration, do you have a solution for configuring gitlab ci/cd so that allure reports can display history like on jenkins and github actions.

I attach the following :
![image](https://github.com/allure-framework/allure2/assets/43233727/be981c81-0fb1-448a-8f6c-48fbd832e18a)

![image](https://github.com/allure-framework/allure2/assets/43233727/e5a76219-94eb-4fec-8366-2d26c93a2428)

![image](https://github.com/allure-framework/allure2/assets/43233727/9cb54fc3-aa7a-4297-bb9a-abf9d3cd7f63)


Thank you sir, I hope you can help me

**Reply by lo0p3r** [reply]
For trend you need to specify the outcoming `report/history` folder from the previous report as the incoming folder of the new report.
My workaround:
1. Make shared folder `//my_share/allure_history` as global ci/cd variable '`CI_ALLURE_HISTORY`'
2. In Allure job set variable: `ALLURE_HISTORY: ${CI_ALLURE_HISTORY}/${CI_PROJECT_NAMESPACE}/${CI_PROJECT_NAME}/${CI_COMMIT_REF_NAME}`
3. In Allure job add this history folder to input: `- allure generate ${ALLURE_HISTORY} ... -o ${ALLURE_REPORT}`
4. In Allure job copy new history to shared folder: `- copy ${ALLURE_REPORT}/history ${ALLURE_HISTORY} -force -recurse`
5. Don't let anyone touch this shared folder :)
